### PR TITLE
added shielding

### PR DESCRIPTION
**Pull request by BenDBrown**
# Context
Added shielding mechanic with visuals and integrated with power/fuel system
# Additions
- Shield png (made by Gabi)
- Shield.cs 
- Shield.tscn
# Modified
- PlayerShip.tscn to add the shield to player prefab
- CentreCalculator to add method that lets me scale the shield easily (this class should probably be renamed)
- PlayerController.cs to add input connection for shielding
- PlayerShip to add shielding integration with fuel and power
# Bug
- AI seems to explode when shielding in front of it

### CODE REVIEW

File: Ship/Scripts/Managers/CenterCalculator.cs [hunk 1]
```diff
@@ -20,6 +21,27 @@ public Vector2 GetGlobalShipCenter(List<Vector2> shipVectors)
...
+		float maxY = float.MinValue;
+		foreach (Vector2 vector in shipVectors)
+		{
+			if(vector.X > maxX) maxX = vector.X;
+			if(vector.X < minX) minX = vector.X;
+			if(vector.Y > maxY) maxY = vector.Y;
+			if(vector.Y < minY) minY = vector.Y;
+		}
+
+		int shieldSpriteSize = 32; // base shield asset is designed to enclose a 32x32 square
```
Comment (GenelleGeerman) [inline Ship/Scripts/Managers/CenterCalculator.cs:41]: Make a constant bbg

File: Ship/Scripts/Shield.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if(node2D is Bullet bullet)
+        {
+            bullet.QueueFree();
+            EmitSignal(SignalName.ShieldHit, powerDraw);
+        }
+    }
+
+    public int GetPowerDraw()
+    {
+        if(Shielding) return powerDraw;
```
Comment (GenelleGeerman) [inline Ship/Scripts/Shield.cs:48]: public int GetPowerDraw() => Shielding? powerdraw : 0;

**Review by GenelleGeerman — APPROVED** [review]

File: Ship/Scripts/Managers/CenterCalculator.cs (hunk 1 above)
Comment (BenDBrown) [reply Ship/Scripts/Managers/CenterCalculator.cs:41]: Yeah, I was thinking this should be a constant in Game.cs as this isn't the first time I've needed to use this value

File: Ship/Scripts/Shield.cs (hunk 2 above)
Comment (BenDBrown) [reply Ship/Scripts/Shield.cs:48]: syntactic sugar, my favourite

**Review by GenelleGeerman — APPROVED** [review]

**Pull request merged**